client/rest-high-level/src/main/java/org/elasticsearch/client/ml/PostDataRequest.java

```diff
@@ -0,0 +1,224 @@
+        this.xContentType = Objects.requireNonNull(xContentType, "content_type must not be null");
+        this.bytesReferences = new ArrayList<>();
+        this.objectMaps = new ArrayList<>();
+    }
+
+    public String getJobId() {
+        return jobId;
+    }
+
+    public String getResetStart() {
+        return resetStart;
+    }
+
+    /**
+     * Specifies the start of the bucket resetting range
+     *
+     * @param resetStart String representation of a timestamp; may be an epoch seconds, epoch millis or an ISO string
+     */
+    public void setResetStart(String resetStart) {
+        this.resetStart = resetStart;
+    }
+
+    public String getResetEnd() {
+        return resetEnd;
+    }
+
+    /**
+     * Specifies the end of the bucket resetting range
+     *
+     * @param resetEnd String representation of a timestamp; may be an epoch seconds, epoch millis or an ISO string
+     */
+    public void setResetEnd(String resetEnd) {
+        this.resetEnd = resetEnd;
+    }
+
+    /**
+     * Gets the transformed content to post.
+     *
+     * This combines both documents added through {@link PostDataRequest#addDoc(BytesReference)} and {@link PostDataRequest#addDoc(Map)}
+     * into a single BytesReference object according to the set XContentType for bulk submission
+     *
+     * If content was set via {@link PostDataRequest#setContent(BytesReference)}, then simply that content is returned.
+     *
+     * @throws IOException on parsing/serialization errors
+     */
+    public BytesReference getContent() throws IOException {
+        if (content != null) {
+            return content;
+        }
+        try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
+            for (BytesReference bytesReference : bytesReferences) {
+                try (StreamInput streamInput = bytesReference.streamInput()) {
+                    builder.rawValue(streamInput, xContentType);
+                }
+            }
+            for (Map<String, Object> objectMap : objectMaps) {
+                builder.map(objectMap);
+            }
+            return BytesReference.bytes(builder);
+        }
+    }
+
+    public XContentType getXContentType() {
+        return xContentType;
+    }
+
+    /**
+     * Set the total content to post.
+     *
+     * @param content BytesReference content to set, format must match the set XContentType
+     */
+    public void setContent(BytesReference content) {
+        this.content = content;
+    }
+
+    /**
+     * Add a document via a ByteReference.
+     *
+     * Ignored if total content is set via {@link PostDataRequest#setContent(BytesReference)}
+     *
+     * @param bytesReference document to add to bulk request, format must match the set XContentType
+     */
+    public void addDoc(BytesReference bytesReference) {
+        this.bytesReferences.add(Objects.requireNonNull(bytesReference, "bytesReferences must not be null"));
```

Comment:
Should we throw an exception if `content != null`?  Or otherwise `return` early to ignore as the Javadoc says.